src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/IUpgradeContext.cs

```diff
@@ -41,5 +42,7 @@ public interface IUpgradeContext : IDisposable
         ValueTask ReloadWorkspaceAsync(CancellationToken token);
 
         IUpgradeContextProperties Properties { get; }
+
+        ICollector<AnalyzeResultDefinition> Results { get; }
```

Comment:
@twsouthwick / @mjrousos how do we feel about this breaking change to the interface? It worries me a bit, but I think it's important enough to warrant and it forces/enables the collection of Upgrade results by default.
I think the only folks it'd break would be those who have forked the repo and implemented their own Contexts...
Alternatively, we could create `IUpgradeContextWithResults : IUpgradeContext` ?

Reply:
I think it's probably fine - just put it as a "Breaking Change". Maybe add a comment to the context that we reserve the right to change it and any custom implementation may break in future releases?